metadata-ingestion/src/datahub/ingestion/source/tableau.py

```diff
@@ -494,27 +679,47 @@ def get_connection_objects(
                 yield obj
 
     def emit_workbooks(self) -> Iterable[MetadataWorkUnit]:
-        projects = (
-            f"projectNameWithin: {json.dumps(self.config.projects)}"
-            if self.config.projects
-            else ""
-        )
+        if self.tableau_project_registry:
```

Comment:
when would this be missing?

Reply:
When no project matches with allow pattern